tests/integration_tests/test_teammatchmaker.py

```diff
@@ -105,6 +111,55 @@ async def test_game_matchmaking(lobby_server):
         assert msg["faction"] == 1
 
 
+@fast_forward(15)
+async def test_game_matchmaking_multiqueue(lobby_server):
+    protos, _ = await connect_players(lobby_server)
+
+    await asyncio.gather(*[
+        read_until_command(proto, "game_info") for proto in protos
+    ])
+
+    await protos[0].send_message({
+        "command": "game_matchmaking",
+        "state": "start",
+        "faction": "uef",
+        "queue_name": "ladder1v1"
+    })
+    await read_until_command(protos[0], "search_info")
+    await asyncio.gather(*[
+        proto.send_message({
+            "command": "game_matchmaking",
+            "state": "start",
+            "faction": "aeon",
+            "queue_name": "tmm2v2"
+        })
+        for proto in protos
+    ])
+    msg = await read_until(
+        protos[0],
+        lambda msg: (
+            msg["command"] == "search_info" and msg["queue_name"] == "ladder1v1"
+        )
+    )
+    assert msg == {
+        "command": "search_info",
+        "queue_name": "ladder1v1",
+        "state": "stop"
+    }
+    msgs = await asyncio.gather(*[client_response(proto) for proto in protos])
+
+    uid = set(msg["uid"] for msg in msgs)
+    assert len(uid) == 1
+    for msg in msgs:
+        assert msg["init_mode"] == 1
+        assert "None" not in msg["name"]
+        assert msg["mod"] == "faf"
+        assert msg["expected_players"] == 4
```

Comment:
So you expect the server to give priority to the tmm2v2 queue, rather than e.g. making two 1v1 matches? Does that maybe make this test a bit brittle in case something in the future changes in which order the queue matches are handled?

Reply:
No, only a single player has queued for ladder, so it's not possible for ladder1v1 to generate a match.

Reply:
Ah, of course.